data_checker/main.resource_policy_check.py

```diff
@@ -0,0 +1,134 @@
+        "AND NOT EXISTS (SELECT 'x' FROM public.workflowitem WHERE " +
+        "public.resourcepolicy.resource_id = public.workflowitem.item_id) " +
+        "AND EXISTS (SELECT 'x' FROM public.item WHERE " +
+        "public.resourcepolicy.resource_id = public.item.item_id) " +
+        "AND epersongroup_id = '0'")
+    # list of tuples
+    result_t = cursor.fetchall()
+    cursor.close()
+    conn.close()
+    # create list from select result
+    for resource_id_t in result_t:
+        resource_id = resource_id_t[0]
+        resource_ids_list.append(resource_id)
+    return resource_ids_list
+
+if __name__ == "__main__":
+    _logger.info('Resource policies checker for anonymous view of items')
+    item_dict_json = "item_dict.json"
+    handle_json = "handle.json"
+    # get a dictionary mapping dspace5 IDs to dspace7 IDs for items
+    item_dict = create_dict_from_json(item_dict_json)
+    # get IDs of item from dspace5 base od select
+    old_item_list = get_data_from_database()
+    # get IDs for dspace7 from IDs from dspace5 based on map
+    new_item_list = convert_old_ids_to_new(old_item_list, item_dict)
+    # list od item IDs from dspace7 which can READ Anonymous
+    item_ids_list=[]
+    # get total pages for search
+    # max page size for this request is 100
+    response = rest_proxy.get('discover/search/objects?sort=score,'
+                              'DESC&size=100&page=0&configuration=default'
+                              '&dsoType=ITEM&embed=thumbnail&embed=item%2Fthumbnail')
+    response_json = convert_response_to_json(response)
+    totalPages = objects = response_json['_embedded']['searchResult']['page']['totalPages']
+    # get result from each page
+    # we don't get items which are withdrawn or discoverable
+    for page in range(totalPages):
+        response = rest_proxy.get('discover/search/objects?sort=score,DESC&size=100&page=' +
+                                  str(page) +
+                                  '&configuration=default&'
+                                  'dsoType=ITEM&embed=thumbnail&embed=item%2Fthumbnail')
+        response_json = convert_response_to_json(response)
+        objects = response_json['_embedded']['searchResult']['_embedded']['objects']
+        # add each object to result list
+        for item in objects:
+            item_ids_list.append(item['_embedded']['indexableObject']['id'])
+
+    # compare expected items in dspace5 and got items from dspace7
+    # log items, which we cannot find
+    item_url = 'core/items'
+    for id_ in new_item_list:
+        if id_ in item_ids_list:
+            item_ids_list.remove(id_)
+        else:
+            # check if we really don't have access to item in Dspace7
+            response = do_api_get_one(item_url, id_)
+            if not response.ok:
+                _logger.error('Do not exist resource policy of anonymous READ for item: ' +
+                              id_ + ' in dspace7!')

```

Comment:
rephrase completely, e.g.: Item (with id: {id} and optionally old handle) is not visible in DSpace7, but it has read resource policy for anonymous in DSpace5 and therefore should be visible!

the old message is almost completely useless for a person who doesn't know about resource policies and hard to understand for a person who knows.

Reply:
if the rephrase is wrong (in meaning), I failed to understand it too :D

Reply:
so should the error message be :  Item (with id: {id} and optionally old handle) is not visible in DSpace7, but it has read resource policy for anonymous in DSpace5 and therefore should be visible ?

Reply:
yes. But consider if it is possible to add handle and or UUID to the message, so that when someone reads it, they can check specific item.

Reply:
..should be visible... in dspace5 or dspace7? because we test if it is visible in dspace7 and this message is only when it is not